Sample/Pluginized/TicTacToe/TicTacToeCore/NeedleGenerated.swift

```diff
@@ -151,3 +121,34 @@ private class LoggedInPluginExtensionProvider: LoggedInPluginExtension {
         loggedInNonCoreComponent = loggedInComponent.nonCoreComponent as! LoggedInNonCoreComponent
     }
 }
+
+
+private func factoryEmptyDependencyProvider(_ component: NeedleFoundation.Scope) -> AnyObject {
+    return EmptyDependencyProvider(component: component)
+}
+
+// MARK: - Registration
+private func registerProviderFactory(_ componentPath: String, _ factory: @escaping (Scope) -> AnyObject) {
+    __DependencyProviderRegistry.instance.registerDependencyProviderFactory(for: componentPath, factory)
+}
+
+private func register1() {
+    registerProviderFactory("^->RootComponent->LoggedOutComponent", factory1434ff4463106e5c4f1bb3a8f24c1d289f2c0f2e)
+    registerProviderFactory("^->RootComponent", factoryEmptyDependencyProvider)
+    registerProviderFactory("^->RootComponent->LoggedInComponent->GameComponent->GameNonCoreComponent->ScoreSheetComponent", factoryb11b7d1dec7e3c9b3dca49b41e44e0ed6a6f8eaf)
+    registerProviderFactory("^->RootComponent->LoggedInComponent->LoggedInNonCoreComponent->ScoreSheetComponent", factory3306c50e89e2421d0b0c65d055996113f3c13de1)
+    registerProviderFactory("^->RootComponent->LoggedInComponent->GameComponent->GameNonCoreComponent", factoryEmptyDependencyProvider)
+    registerProviderFactory("^->RootComponent->LoggedInComponent->LoggedInNonCoreComponent", factoryEmptyDependencyProvider)
+    registerProviderFactory("^->RootComponent->LoggedInComponent->GameComponent", factorycf9c02c4def4e3d508816cd03d3cf415b70dfb0e)
+    registerProviderFactory("^->RootComponent->LoggedInComponent", factoryEmptyDependencyProvider)
+    __PluginExtensionProviderRegistry.instance.registerPluginExtensionProviderFactory(for: "GameComponent") { component in
```

Comment:
Do PluginExtensions also need to remove the closures and be replaced with a named function?

Reply:
At least in the examples I've seen there isn't the proliferation of calls. For instance I ran into over 10k of the register calls, but only around 20-30 of the plugin extensions. Granted, I know very little about Needle, so I'm not sure if that pattern is true on other codebases.

Reply:
Right, the key is to notice that there's no path involved, hence no exponential explosion as the tree gets deep. So yes, the number should remain small enough even in the future to ignore these for your optimization.